src/cmdstan/stansummary_helper.hpp

```diff
@@ -190,31 +190,434 @@ int matrix_index(std::vector<int> &index, const std::vector<int> &dims) {
-                     << "integers between 0 and 100, increasing, found: "
-                     << percentiles_spec << ".";
-      throw std::domain_error(message_stream.str());
+      message_stream << "value(s) for option --percentiles must be "
+                     << "integers between 0 and 100, strictly increasing."
+                     << std::endl;
+      throw boost::program_options::error(message_stream.str());
     }
     probs[i] = pct * 1.0 / 100.0;
   }
   return probs;
 }
 
+stan::mcmc::chains<> parse_csv_files(const std::vector<std::string> &filenames,
+                                     stan::io::stan_csv_metadata &metadata,
+                                     Eigen::VectorXd &warmup_times,
+                                     Eigen::VectorXd &sampling_times,
+                                     Eigen::VectorXi &thin, std::ostream *out) {
+  // instantiate stan::mcmc::chains object by parsing first file
+  std::ifstream ifstream;
+  ifstream.open(filenames[0].c_str());
+  stan::io::stan_csv stan_csv = stan::io::stan_csv_reader::parse(ifstream, out);
+  warmup_times(0) = stan_csv.timing.warmup;
+  sampling_times(0) = stan_csv.timing.sampling;
+  stan::mcmc::chains<> chains(stan_csv);
+  ifstream.close();
+  thin(0) = stan_csv.metadata.thin;
+  metadata = stan_csv.metadata;
+
+  // parse rest of input files, add to chains
+  for (std::vector<std::string>::size_type chain = 1; chain < filenames.size();
+       chain++) {
+    ifstream.open(filenames[chain].c_str());
+    stan_csv = stan::io::stan_csv_reader::parse(ifstream, out);
+    chains.add(stan_csv);
+    ifstream.close();
+    thin(chain) = stan_csv.metadata.thin;
+    warmup_times(chain) = stan_csv.timing.warmup;
+    sampling_times(chain) = stan_csv.timing.sampling;
+  }
+  return chains;
+}
+
+std::vector<std::string> get_sampler_params_header(
+    const std::vector<std::string> &percentiles) {
+  // Mean, StdDev, ... percentiles ...
+  std::vector<std::string> sampler_params_header(percentiles.size() + 2);
+  sampler_params_header.at(0) = "Mean";
+  sampler_params_header.at(1) = "StdDev";
+  size_t offset = 2;
+  for (size_t i = 0; i < percentiles.size(); ++i) {
+    sampler_params_header[i + offset] = percentiles[i] + '%';
+  }
+  return sampler_params_header;
+}
+
+std::vector<std::string> get_model_params_header(
+    const std::vector<std::string> &percentiles) {
+  // Mean, MCSE,  StdDev, ... percentiles ..., N_eff, N_eff/s, R_hat
+  std::vector<std::string> model_params_header(percentiles.size() + 6);
+  model_params_header.at(0) = "Mean";
+  model_params_header.at(1) = "MCSE";
+  model_params_header.at(2) = "StdDev";
+  size_t offset = 3;
+  for (size_t i = 0; i < percentiles.size(); ++i) {
+    model_params_header[i + offset] = percentiles[i] + '%';
+  }
+  offset += percentiles.size();
+  model_params_header.at(offset) = "N_Eff";
+  model_params_header.at(offset + 1) = "N_Eff/s";
+  model_params_header.at(offset + 2) = "R_hat";
```

Comment:
```suggestion
  for (size_t i = 0; i < percentiles.size(); ++i) {
    model_params_header[i + 3] = percentiles[i] + '%';
  }
  const size_t offset = 3 + percentiles.size();
  model_params_header.at(offset) = "N_Eff";
  model_params_header.at(offset + 1) = "N_Eff/s";
  model_params_header.at(offset + 2) = "R_hat";
```

Reply:
implemented - omitted `const` - doesn't do anything, code-wise, and not consistent with coding style.